Libraries/Core/Devtools/symbolicateStackTrace.js

```diff
@@ -18,27 +18,38 @@ const {fetch} = require('fetch');
 
 import type {StackFrame} from 'parseErrorStack';
 
+function isSourcedFromDisk(sourcePath: string): boolean {
+  return !/^http/.test(sourcePath) && /[\\/]/.test(sourcePath);
+}
+
 async function symbolicateStackTrace(stack: Array<StackFrame>): Promise<Array<StackFrame>> {
   const devServer = getDevServer();
   if (!devServer.bundleLoadedFromServer) {
     throw new Error('Bundle was not loaded from the packager');
   }
-  if (SourceCode.scriptURL) {
-    for (let i = 0; i < stack.length; ++i) {
+
+  let stackCopy = stack;
+
+  if (SourceCode.scriptURL && stack.length > 0 && isSourcedFromDisk(stack[0].file)) {
+    let foundInternalSource: boolean = false;
+    stackCopy = stack.map((frame: StackFrame) => {
       // If the sources exist on disk rather than appearing to come from the packager,
       // replace the location with the packager URL until we reach an internal source
       // which does not have a path (no slashes), indicating a switch from within
       // the application to a surrounding debugging environment.
-      if (/^http/.test(stack[i].file) || !/[\\/]/.test(stack[i].file)) {
-        break;
+      if (!foundInternalSource && isSourcedFromDisk(frame.file)) {
+        // Copy frame into new object and replace 'file' property
+        return Object.assign({}, frame, {file: SourceCode.scriptURL});
```

Comment:
fyi: this can be done with JS syntax and one fewer object creation: `return {...frame, file: SourceCode.scriptURL}` (just a style thing more than a performance thing)